src/vs/code/electron-main/protocol.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Event } from 'vs/base/common/event';
+import { Disposable, DisposableStore, IDisposable, toDisposable } from 'vs/base/common/lifecycle';
+import { FileAccess, Schemas } from 'vs/base/common/network';
+import { URI } from 'vs/base/common/uri';
+import { INativeEnvironmentService } from 'vs/platform/environment/common/environment';
+import { session } from 'electron';
+import { ILogService } from 'vs/platform/log/common/log';
+import { TernarySearchTree } from 'vs/base/common/map';
+import { isLinux } from 'vs/base/common/platform';
+import { IWindowsMainService } from 'vs/platform/windows/electron-main/windows';
+
+type ProtocolCallback = { (result: string | Electron.FilePathWithHeaders | { error: number }): void };
+
+export class FileProtocolHandler extends Disposable {
+
+	private readonly validRoots = TernarySearchTree.forUris<boolean>(() => !isLinux);
+
+	constructor(
+		@INativeEnvironmentService environmentService: INativeEnvironmentService,
+		@ILogService private readonly logService: ILogService
+	) {
+		super();
+
+		const { defaultSession } = session;
+
+		// Define an initial set of roots we allow loading from
+		// - appRoot	: all files installed as part of the app
+		// - extensions : all files shipped from extensions
+		this.validRoots.set(URI.file(environmentService.appRoot), true);
+		this.validRoots.set(URI.file(environmentService.extensionsPath), true);
+
+		// Register vscode-file:// handler
+		defaultSession.protocol.registerFileProtocol(Schemas.vscodeFileResource, (request, callback) => this.handleResourceRequest(request, callback as unknown as ProtocolCallback));
+
+		// Cleanup
+		this._register(toDisposable(() => {
+			defaultSession.protocol.unregisterProtocol(Schemas.vscodeFileResource);
+			defaultSession.protocol.uninterceptProtocol(Schemas.file);
```

Comment:
there is no corresponding interceptProtocol call in this PR.

Reply:
Thanks, will remove.